From 7583c92b266cd632712ced40be837f52a8de9715 Mon Sep 17 00:00:00 2001
From: Kay Sievers <kay@vrfy.org>
Date: Fri, 11 Oct 2013 11:23:34 +0200
Subject: [PATCH] udev: usb_id - remove obsoleted bInterfaceSubClass == 5 match

bInterfaceSubClass == 5 is not a "floppy"; just identify the obsolete
QIC-157 interface as "generic".
---
 src/udev/udev-builtin-usb_id.c | 7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/src/udev/udev-builtin-usb_id.c b/src/udev/udev-builtin-usb_id.c
index e3bbd05..3e2f43e 100644
--- a/src/udev/udev-builtin-usb_id.c
+++ b/src/udev/udev-builtin-usb_id.c
@@ -91,6 +91,9 @@ static int set_usb_mass_storage_ifsubtype(char *to, const char *from, size_t len
         type_num = strtoul(from, &eptr, 0);
         if (eptr != from) {
                 switch (type_num) {
+                case 1: /* RBC devices */
+                        type = "rbc";
+                        break;
                 case 2:
                         type = "atapi";
                         break;
@@ -98,12 +101,8 @@ static int set_usb_mass_storage_ifsubtype(char *to, const char *from, size_t len
                         type = "tape";
                         break;
                 case 4: /* UFI */
-                case 5: /* SFF-8070i */
                         type = "floppy";
                         break;
-                case 1: /* RBC devices */
-                        type = "rbc";
-                        break;
                 case 6: /* Transparent SPC-2 devices */
                         type = "scsi";
                         break;
